File: src/main/java/com/evolutions/database/Mode.java
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package com.evolutions.database;

/** Application mode, either `DEV`, `TEST`, or `PROD`. */
public enum Mode {
  DEV,
  TEST,
  PROD;

  public com.evolutions.database.teste.Mode asScala() {
    if (this == DEV) {
      return com.evolutions.database.teste.Mode.Dev$.MODULE$;
    } else if (this == PROD) {
      return com.evolutions.database.teste.Mode.Prod$.MODULE$;
    }
    return com.evolutions.database.teste.Mode.Test$.MODULE$;
  }
}

File: src/main/scala/com/evolutions/database/teste/Mode.scala
package com.evolutions.database.teste

/**
 * Application mode, either `Dev`, `Test`, or `Prod`.
 *
 * @see [[com.evolutions.database.Mode]]
 */
sealed abstract class Mode(val asJava: com.evolutions.database.Mode)

object Mode {
  case object Dev  extends Mode(com.evolutions.database.Mode.DEV)
  case object Test extends Mode(com.evolutions.database.Mode.TEST)
  case object Prod extends Mode(com.evolutions.database.Mode.PROD)

  lazy val values: Set[com.evolutions.database.teste.Mode] = Set(Dev, Test, Prod)
}

